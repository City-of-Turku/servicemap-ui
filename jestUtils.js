import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });