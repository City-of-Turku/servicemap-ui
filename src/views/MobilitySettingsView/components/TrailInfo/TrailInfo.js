import { Typography } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';

const TrailInfo = ({ classes, intl, item }) => {
  const renderLength = () => (
    <Typography
      component="p"
      variant="body2"
      aria-label={intl.formatMessage(
        { id: 'mobilityPlatform.menu.markedTrails.length' },
        { value: item.extra.length_km },
      )}
    >
      {intl.formatMessage(
        { id: 'mobilityPlatform.menu.markedTrails.length' },
        { value: item.extra.length_km },
      )}
    </Typography>
  );

  return (
    <div className={classes.container}>
      <div className={classes.paragraph}>{renderLength()}</div>
    </div>
  );
};

TrailInfo.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  item: PropTypes.objectOf(PropTypes.any),
};

TrailInfo.defaultProps = {
  item: {},
};

export default TrailInfo;
