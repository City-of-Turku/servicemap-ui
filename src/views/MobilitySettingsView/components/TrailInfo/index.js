import TrailInfo from './TrailInfo';

export default TrailInfo;
