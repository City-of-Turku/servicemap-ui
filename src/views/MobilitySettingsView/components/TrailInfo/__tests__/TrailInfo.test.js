// Link.react.test.js
import React from 'react';
import TrailInfo from '../index';
import { getRenderWithProviders } from '../../../../../../jestUtils';

const mockProps = {
  item: {
    extra: {
      length_km: 5,
    },
  },
};

const renderWithProviders = getRenderWithProviders({});

describe('<TrailInfo />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<TrailInfo {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<TrailInfo {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toEqual('Reitin pituus: 5 km');
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<TrailInfo {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toEqual('Reitin pituus: 5 km');
  });
});
