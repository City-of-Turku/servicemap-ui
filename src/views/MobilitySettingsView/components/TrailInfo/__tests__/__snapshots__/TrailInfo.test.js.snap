// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrailInfo /> should match snapshot 1`] = `
<div>
  <div>
    <div
      class="injectIntl(TrailInfo)-paragraph-1"
    >
      <p
        aria-label="Reitin pituus: 5 km"
        class="MuiTypography-root MuiTypography-body2"
      >
        Reitin pituus: 5 km
      </p>
    </div>
  </div>
</div>
`;
